zimui/src/pages/ChallengePage.vue

```diff
@@ -1,126 +1,55 @@
-  solution.value = challenge.value.seed || ''
-}
+const superblock = computed(() => singlePathParam(route.params.superblock))
+const course = computed(() => singlePathParam(route.params.course))
+const slug = computed(() => singlePathParam(route.params.slug))
 
 watch(
-  () => route.params,
-  async (p: RouteParams) => {
-    await updateChallenge(p)
-  }
+  () => `${superblock.value}/${course.value}`,
+  () => {
+    main.fetchMeta(superblock.value, course.value)
+  },
+  { immediate: true }
 )
 
-const onReset = () => {
-  solution.value = challenge.value.seed || ''
-}
-
-const onSetSolution = () => {
-  solution.value = challenge.value.solutions[0]
-}
-
-const nextChallenge: ComputedRef<ChallengeJSON | null> = computed(() => {
-  if (!challenge.value) return null
-  const challengeIndex = challenges.value.findIndex((c) => c.slug === params.value.slug)
-  if (challengeIndex <= challenges.value.length) {
-    return challenges.value[challengeIndex + 1]
-  }
-  return null
-})
-
-const nextChallengeLink: ComputedRef<{ title: string; url: string } | undefined> = computed(() => {
-  if (nextChallenge.value) {
-    return {
-      title: nextChallenge.value.title,
-      url: `/${params.value.superblock}/${params.value.course}/${nextChallenge.value.slug}`
-    }
-  }
-  return undefined
-})
-
-await updateChallenge(params.value)
+watch(
+  () => `${superblock.value}/${course.value}/${slug.value}`,
+  () => {
+    main.fetchChallenge(superblock.value, course.value, slug.value)
+  },
+  { immediate: true }
+)
 </script>
 
 <template>
-  <div v-if="['1', '4', '5'].includes(challenge.header['challengeType'])" class="split">
-    <div class="left">
-      <ChallengeInstructions
-        :title="challenge.header['title']"
-        :instructions="challenge.instructions || ''"
-        :description="challenge.description"
-        :coursetitle="locales.blocks[params.course as string].title"
-        :coursepath="`/${params.superblock}/${params.course}`"
-        :curriculumtitle="locales.title"
-      ></ChallengeInstructions>
-      <ChallengeRunner
-        :challenge="challenge"
-        :solution="solution || ''"
-        :next-challenge="nextChallengeLink"
-        @reset="onReset"
-        @set-solution="onSetSolution"
-        @logs="(value: any) => (logs = value)"
-      ></ChallengeRunner>
-    </div>
-    <div class="right">
-      <CodeEditor
-        :source-code="solution"
-        class="codeEditor"
-        @update="(event: any) => (solution = event)"
-      ></CodeEditor>
-      <ConsoleLogger class="console" :logs="logs || []"></ConsoleLogger>
-    </div>
-  </div>
-
-  <div v-else class="unsupported-page">
-    <div class="unsupported-container">
-      <div class="unsupported-content">
-        <p class="unsupported-text">This type of challenge is not yet working offline.</p>
+  <div v-if="main.challenge && main.locales">
+    <div v-if="['1', '4', '5'].includes(main.challenge.header['challengeType'])" class="split">
```

Comment:
I know it was already there but I think those magic values in template makes it harder to maintain